src/constraint_system/composer.rs

```diff
@@ -1159,6 +1188,115 @@ impl StandardComposer {
         self.logic_gate(a, b, num_bits, false)
     }
 
+    /// Creates an add gate for specific bases. This function will add an X-coordinate into
+    /// an accumulator for each round of scalar multiplication.
+    pub fn fixed_base_add(&mut self, a: Fr, b: Fr, c: Fr, d: Fr, num_bits: usize) {
```

Comment:
There's not any loop  or recursion in this function. 

The whole addition of values inside of the program memory should be done here instead of other places like `scalar_mul.rs`